src/bookmarks/bookmarkfiltermodel.cpp

```diff
@@ -0,0 +1,30 @@
+#include "bookmarkfiltermodel.h"
+
+BookmarkFilterModel::BookmarkFilterModel(QObject *parent)
+    : QSortFilterProxyModel(parent)
+{
+
+}
+
+int BookmarkFilterModel::getIndex(int currentIndex)
+{
+    QModelIndex proxyIndex = index(currentIndex, 0);
+    QModelIndex sourceIndex = mapToSource(proxyIndex);
+    return sourceIndex.row();
+}
+
+bool BookmarkFilterModel::filterAcceptsRow(int sourceRow, const QModelIndex &sourceParent) const
+{
+    bool contains = false;
+    QModelIndex index = sourceModel()->index(sourceRow, 0, sourceParent);
+
+    for (int i = 0; i < sourceModel()->roleNames().count(); i++) {
+        int it = i + static_cast<int>(Qt::UserRole);
+        QString roleName = sourceModel()->roleNames().value(it);
+        if (roleName == "url" || roleName == "title") {
+            if (sourceModel()->data(index, it).toString().trimmed().contains(filterRegExp().pattern(), Qt::CaseInsensitive))
```

Comment:
Why `filterRegExp().pattern()`?
By the way, does it mean that user has provided the search string as a regexp pattern string (instead of the wildcard one)?